shared-library.cmake

```diff
@@ -24,6 +24,19 @@ IF(WIN32)
 ELSEIF(UNIX)
   SET(LIBDIR_KW "-L")
   SET(LIBINCL_KW "-l")
-  SET(LIBINCL_ABSKW "-l:")
   SET(LIB_EXT "")
+
+  # Using -l:/some/absolute/path.so was an "undocumented ld feature, in
+  # actual fact a ld bug, that has since been fixed".
+  # This was apparently used (e.g. in ROS) because of pkg-config problems that
+  # have since been fixed.
+  # See: https://github.com/ros/catkin/issues/694#issuecomment-88323282
+  # Note: ld version on Linux can be 2.25.1 or 2.24
+  EXECUTE_PROCESS(COMMAND ${CMAKE_LINKER} -v OUTPUT_VARIABLE LD_VERSION_STR)
+  STRING(REGEX MATCH "([0-9]+.[0-9]+.?[0-9]*)" LD_VERSION ${LD_VERSION_STR})
```

Comment:
I suggest:
([0-9]+\\.[0-9]+(\\.[0-9]+)?)
This parses the version correctly for both the GNU Linker and OSX linker (which is version 253.6 on Xcode 7.0 and works as ld > 2.24.90 w.r.t -l: support